src/gtk/textctrl.cpp

```diff
@@ -1017,6 +1023,52 @@ void wxTextCtrl::GTKSetJustification()
     }
 }
 
+#if wxUSE_SPELLCHECK
+
+bool wxTextCtrl::EnableSpellCheck(const bool enable)
+{
+    wxCHECK_MSG( IsMultiLine(), false, 
+                "Unable to enable spell check on control "
+                "which does not have wxTE_MULTILINE style" );
+
+    GtkTextView *textview = GTK_TEXT_VIEW(m_text);
+    wxCHECK_MSG( textview, false, wxS("wxTextCtrl is not a GtkTextView" ));
+
+    wxString lang = wxLocale::GetLanguageCanonicalName(wxLANGUAGE_DEFAULT);
+
+    GtkSpellChecker *spell = gtk_spell_checker_get_from_text_view(textview);
+
+    if ( enable )
+    {
+        if ( !spell )
+        {
+            spell = gtk_spell_checker_new();
+            gtk_spell_checker_attach(spell, textview);
+        }
+
+        if ( !gtk_spell_checker_set_language(spell, NULL, NULL) )
+            return false;
+    }
+    else
+    {
+        if ( spell )
+            gtk_spell_checker_detach(spell);
```

Comment:
Shouldn't we destroy (i.e. unref) it here?

Reply:
I don't think that it is necessary according to this page: http://gtkspell.sourceforge.net/doc/gtkspell3/tutorial-advanced.html

It looks like g_object_unref() is connected to the destroy signal and it is necessary to explicitly take another reference if you need to keep it?